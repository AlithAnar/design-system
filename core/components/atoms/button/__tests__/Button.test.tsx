import * as React from 'react';
import { shallow } from 'enzyme';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';
import Button, { ButtonProps as Props } from '../Button';

const BooleanValue = [true, false];
const buttonType = ['button', 'submit', 'reset'];
const icon = 'events';
const iconAlign = ['left', 'right'];
const size = ['tiny', 'regular', 'large'];
const appearance = ['basic', 'primary', 'success', 'alert', 'transparent'];

describe('Button component', () => {
  const mapper: Record<string, any> = {
    type: valueHelper(buttonType, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <Button
          {...attr}
        >
          Button
        </Button >
      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Button component', () => {
  const mapper: Record<string, any> = {
    appearance: valueHelper(appearance, { required: true, iterate: true }),
    expanded: valueHelper(BooleanValue, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <Button
          {...attr}
        >
          Button
        </Button >
      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Button component', () => {
  const mapper: Record<string, any> = {
    size: valueHelper(size, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <Button
          {...attr}
        >
          Button
        </Button >
      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Button component', () => {
  const mapper: Record<string, any> = {
    size: valueHelper(size, { required: true, iterate: true }),
    icon: valueHelper(icon, { required: true }),
    iconAlign: valueHelper(iconAlign, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <Button
          {...attr}
        >
          Button
        </Button >
      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Button component', () => {
  const mapper: Record<string, any> = {
    appearance: valueHelper(appearance, { required: true, iterate: true }),
    disabled: valueHelper(true, { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <Button
          {...attr}
        >
          Button
        </Button >
      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Button component', () => {
  const mapper: Record<string, any> = {
    appearance: valueHelper(appearance, { required: true, iterate: true }),
    selected: valueHelper(true, { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <Button
          {...attr}
        >
          Button
        </Button >
      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Button component', () => {
  const mapper: Record<string, any> = {
    appearance: valueHelper(appearance, { required: true, iterate: true }),
    loading: valueHelper(true, { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <Button
          {...attr}
        >
          Button
        </Button >
      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Button component with no children', () => {
  const mapper: Record<string, any> = {
    appearance: valueHelper(appearance, { required: true, iterate: true }),
    loading: valueHelper(true, { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <Button
          {...attr}
        />
      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});
