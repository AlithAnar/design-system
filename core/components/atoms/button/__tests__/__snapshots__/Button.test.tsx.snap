// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component 
  appearance: "alert", disabled: true
 1`] = `
<button
  className="Button Button--regular Button--alert Button--iconAlign-left"
  disabled={true}
  tabIndex={0}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "alert", expanded: false
 1`] = `
<button
  className="Button Button--regular Button--alert Button--iconAlign-left"
  tabIndex={0}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "alert", expanded: true
 1`] = `
<button
  className="Button Button--expanded Button--regular Button--alert Button--iconAlign-left"
  tabIndex={0}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "alert", loading: true
 1`] = `
<button
  className="Button Button--regular Button--alert Button--iconAlign-left"
  disabled={true}
  tabIndex={0}
>
  <span
    className="Button-spinner Button-spinner--left"
  >
    <Spinner
      appearance="white"
      size="small"
    />
  </span>
  Button
</button>
`;

exports[`Button component 
  appearance: "alert", selected: true
 1`] = `
<button
  className="Button Button--regular Button--alert Button--iconAlign-left"
  tabIndex={0}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "basic", disabled: true
 1`] = `
<button
  className="Button Button--regular Button--basic Button--iconAlign-left"
  disabled={true}
  tabIndex={0}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "basic", expanded: false
 1`] = `
<button
  className="Button Button--regular Button--basic Button--iconAlign-left"
  tabIndex={0}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "basic", expanded: true
 1`] = `
<button
  className="Button Button--expanded Button--regular Button--basic Button--iconAlign-left"
  tabIndex={0}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "basic", loading: true
 1`] = `
<button
  className="Button Button--regular Button--basic Button--iconAlign-left"
  disabled={true}
  tabIndex={0}
>
  <span
    className="Button-spinner Button-spinner--left"
  >
    <Spinner
      appearance="secondary"
      size="small"
    />
  </span>
  Button
</button>
`;

exports[`Button component 
  appearance: "basic", selected: true
 1`] = `
<button
  className="Button Button--regular Button--basic Button--selected Button--iconAlign-left"
  tabIndex={0}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "primary", disabled: true
 1`] = `
<button
  className="Button Button--regular Button--primary Button--iconAlign-left"
  disabled={true}
  tabIndex={0}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "primary", expanded: false
 1`] = `
<button
  className="Button Button--regular Button--primary Button--iconAlign-left"
  tabIndex={0}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "primary", expanded: true
 1`] = `
<button
  className="Button Button--expanded Button--regular Button--primary Button--iconAlign-left"
  tabIndex={0}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "primary", loading: true
 1`] = `
<button
  className="Button Button--regular Button--primary Button--iconAlign-left"
  disabled={true}
  tabIndex={0}
>
  <span
    className="Button-spinner Button-spinner--left"
  >
    <Spinner
      appearance="white"
      size="small"
    />
  </span>
  Button
</button>
`;

exports[`Button component 
  appearance: "primary", selected: true
 1`] = `
<button
  className="Button Button--regular Button--primary Button--iconAlign-left"
  tabIndex={0}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "success", disabled: true
 1`] = `
<button
  className="Button Button--regular Button--success Button--iconAlign-left"
  disabled={true}
  tabIndex={0}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "success", expanded: false
 1`] = `
<button
  className="Button Button--regular Button--success Button--iconAlign-left"
  tabIndex={0}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "success", expanded: true
 1`] = `
<button
  className="Button Button--expanded Button--regular Button--success Button--iconAlign-left"
  tabIndex={0}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "success", loading: true
 1`] = `
<button
  className="Button Button--regular Button--success Button--iconAlign-left"
  disabled={true}
  tabIndex={0}
>
  <span
    className="Button-spinner Button-spinner--left"
  >
    <Spinner
      appearance="white"
      size="small"
    />
  </span>
  Button
</button>
`;

exports[`Button component 
  appearance: "success", selected: true
 1`] = `
<button
  className="Button Button--regular Button--success Button--iconAlign-left"
  tabIndex={0}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "transparent", disabled: true
 1`] = `
<button
  className="Button Button--regular Button--transparent Button--iconAlign-left"
  disabled={true}
  tabIndex={0}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "transparent", expanded: false
 1`] = `
<button
  className="Button Button--regular Button--transparent Button--iconAlign-left"
  tabIndex={0}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "transparent", expanded: true
 1`] = `
<button
  className="Button Button--expanded Button--regular Button--transparent Button--iconAlign-left"
  tabIndex={0}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "transparent", loading: true
 1`] = `
<button
  className="Button Button--regular Button--transparent Button--iconAlign-left"
  disabled={true}
  tabIndex={0}
>
  <span
    className="Button-spinner Button-spinner--left"
  >
    <Spinner
      appearance="secondary"
      size="small"
    />
  </span>
  Button
</button>
`;

exports[`Button component 
  appearance: "transparent", selected: true
 1`] = `
<button
  className="Button Button--regular Button--transparent Button--selected Button--iconAlign-left"
  tabIndex={0}
>
  Button
</button>
`;

exports[`Button component 
  size: "large"
 1`] = `
<button
  className="Button Button--large Button--basic Button--iconAlign-left"
  tabIndex={0}
>
  Button
</button>
`;

exports[`Button component 
  size: "large", icon: "events", iconAlign: "left"
 1`] = `
<button
  className="Button Button--large Button--basic Button--iconAlign-left"
  tabIndex={0}
>
  <div
    className="Button-icon Button-icon--left"
  >
    <Icon
      appearance="default"
      name="events"
      size={20}
    />
  </div>
  Button
</button>
`;

exports[`Button component 
  size: "large", icon: "events", iconAlign: "right"
 1`] = `
<button
  className="Button Button--large Button--basic Button--iconAlign-right"
  tabIndex={0}
>
  <div
    className="Button-icon Button-icon--right"
  >
    <Icon
      appearance="default"
      name="events"
      size={20}
    />
  </div>
  Button
</button>
`;

exports[`Button component 
  size: "regular"
 1`] = `
<button
  className="Button Button--regular Button--basic Button--iconAlign-left"
  tabIndex={0}
>
  Button
</button>
`;

exports[`Button component 
  size: "regular", icon: "events", iconAlign: "left"
 1`] = `
<button
  className="Button Button--regular Button--basic Button--iconAlign-left"
  tabIndex={0}
>
  <div
    className="Button-icon Button-icon--left"
  >
    <Icon
      appearance="default"
      name="events"
      size={16}
    />
  </div>
  Button
</button>
`;

exports[`Button component 
  size: "regular", icon: "events", iconAlign: "right"
 1`] = `
<button
  className="Button Button--regular Button--basic Button--iconAlign-right"
  tabIndex={0}
>
  <div
    className="Button-icon Button-icon--right"
  >
    <Icon
      appearance="default"
      name="events"
      size={16}
    />
  </div>
  Button
</button>
`;

exports[`Button component 
  size: "tiny"
 1`] = `
<button
  className="Button Button--tiny Button--basic Button--iconAlign-left"
  tabIndex={0}
>
  Button
</button>
`;

exports[`Button component 
  size: "tiny", icon: "events", iconAlign: "left"
 1`] = `
<button
  className="Button Button--tiny Button--basic Button--iconAlign-left"
  tabIndex={0}
>
  <div
    className="Button-icon Button-icon--left"
  >
    <Icon
      appearance="default"
      name="events"
      size={12}
    />
  </div>
  Button
</button>
`;

exports[`Button component 
  size: "tiny", icon: "events", iconAlign: "right"
 1`] = `
<button
  className="Button Button--tiny Button--basic Button--iconAlign-right"
  tabIndex={0}
>
  <div
    className="Button-icon Button-icon--right"
  >
    <Icon
      appearance="default"
      name="events"
      size={12}
    />
  </div>
  Button
</button>
`;

exports[`Button component 
  type: "button"
 1`] = `
<button
  className="Button Button--regular Button--basic Button--iconAlign-left"
  tabIndex={0}
  type="button"
>
  Button
</button>
`;

exports[`Button component 
  type: "reset"
 1`] = `
<button
  className="Button Button--regular Button--basic Button--iconAlign-left"
  tabIndex={0}
  type="reset"
>
  Button
</button>
`;

exports[`Button component 
  type: "submit"
 1`] = `
<button
  className="Button Button--regular Button--basic Button--iconAlign-left"
  tabIndex={0}
  type="submit"
>
  Button
</button>
`;

exports[`Button component with no children 
  appearance: "alert", loading: true
 1`] = `
<button
  className="Button Button--regular Button--regularSquare Button--alert"
  disabled={true}
  tabIndex={0}
>
  <span
    className="Button-spinner"
  >
    <Spinner
      appearance="white"
      size="small"
    />
  </span>
</button>
`;

exports[`Button component with no children 
  appearance: "basic", loading: true
 1`] = `
<button
  className="Button Button--regular Button--regularSquare Button--basic"
  disabled={true}
  tabIndex={0}
>
  <span
    className="Button-spinner"
  >
    <Spinner
      appearance="secondary"
      size="small"
    />
  </span>
</button>
`;

exports[`Button component with no children 
  appearance: "primary", loading: true
 1`] = `
<button
  className="Button Button--regular Button--regularSquare Button--primary"
  disabled={true}
  tabIndex={0}
>
  <span
    className="Button-spinner"
  >
    <Spinner
      appearance="white"
      size="small"
    />
  </span>
</button>
`;

exports[`Button component with no children 
  appearance: "success", loading: true
 1`] = `
<button
  className="Button Button--regular Button--regularSquare Button--success"
  disabled={true}
  tabIndex={0}
>
  <span
    className="Button-spinner"
  >
    <Spinner
      appearance="white"
      size="small"
    />
  </span>
</button>
`;

exports[`Button component with no children 
  appearance: "transparent", loading: true
 1`] = `
<button
  className="Button Button--regular Button--regularSquare Button--transparent"
  disabled={true}
  tabIndex={0}
>
  <span
    className="Button-spinner"
  >
    <Spinner
      appearance="secondary"
      size="small"
    />
  </span>
</button>
`;
